hedera-mirror-rest/utils.js

```
@@ -724,6 +733,22 @@ const parsePublicKey = (publicKey) => {
   return decodedKey == null ? publicKey : decodedKey;
 };
 
+const getTransactionTypeQuery = (parsedQueryParams) => {
+  if (_.isNil(parsedQueryParams)) {
+    return '';
+  }
+
+  const transactionType = parsedQueryParams[constants.filterKeys.TRANSACTION_TYPE];
+
+  if (isValidTransactionType(transactionType)) {
+    return `${constants.transactionColumns.TYPE}${opsMap.eq}${
+      constants.transactionTypes[transactionType.toUpperCase()]
+    }`;
+  }
+
+  return '';
```

I don't think it's a good user experience to silently fail and return all transactions if they mistype the transaction type. We should return an error if it's not one of the known types so they can adjust their criteria. Add a test for this scenario as well.

---

Good point.
Will throw